tests/ci/cdk/cdk/aws_lc_github_ci_x509_stack.py

```diff
@@ -1,25 +1,44 @@
-from aws_cdk import Duration, Stack, aws_codebuild as codebuild, aws_s3 as s3
+import typing
+
+from aws_cdk import (
+    Duration,
+    Stack,
+    aws_codebuild as codebuild,
+    aws_s3 as s3,
+    Environment,
+)
 from constructs import Construct
 from util.build_spec_loader import BuildSpecLoader
 from util.metadata import (
     GITHUB_PUSH_CI_BRANCH_TARGETS,
     GITHUB_REPO_NAME,
     GITHUB_REPO_OWNER,
+    PRE_PROD_ACCOUNT,
+    STAGING_GITHUB_REPO_OWNER,
+    STAGING_GITHUB_REPO_NAME,
 )
 
-
 class AwsLcGitHubX509CIStack(Stack):
     def __init__(
         self,
         scope: Construct,
         id: str,
+        env: typing.Union[Environment, typing.Dict[str, typing.Any]],
         **kwargs,
     ) -> None:
-        super().__init__(scope, id, **kwargs)
+        super().__init__(scope, id, env=env, **kwargs)
+
+        github_repo_owner = GITHUB_REPO_OWNER
+        github_repo_name = GITHUB_REPO_NAME
+
+        if env.account == PRE_PROD_ACCOUNT:
+            github_repo_owner = STAGING_GITHUB_REPO_OWNER
+            github_repo_name = STAGING_GITHUB_REPO_NAME
```

Comment:
At this point this check is repeated in a number of components, should this be passed into the constructor already setup correct?